File: CS/LizardKingStudios.Services/LizardKingStudios.Services.SpellCheck/userConfirm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using Microsoft.Office.Interop.Word;
using System.Windows.Forms;

namespace LizardKingStudios.Services.SpellCheck
{
    public partial class userConfirm : Form
    {
        private SpellingSuggestions _suggestionsList = null;
        private string _selectedValue = String.Empty;

        public userConfirm()
        {
            InitializeComponent();
        }

        public SpellingSuggestions suggestionsList
        {
            set
            {
                _suggestionsList = value;

                loadSuggestionsList();
            }
        }

        public string selectedValue
        {
            get
            {
                return _selectedValue;
            }
        }

        private void loadSuggestionsList()
        {
            if (_suggestionsList!=null)
            {
                foreach (SpellingSuggestion theSuggestion in _suggestionsList)
                {
                    Suggestions.Items.Add(theSuggestion.Name);
                }

                Suggestions.SelectedIndex = 0;
            }
        }

        private void selectSuggestion_Click(object sender, EventArgs e)
        {
            if (Suggestions.SelectedIndex==-1)
            {
                MessageBox.Show("You must select a suggestion first!");
            }
            else
            {
                _selectedValue = Suggestions.SelectedItem.ToString();

                MessageBox.Show("selected: " + _selectedValue);
                this.Close();
            }
        }

        private void cancelForm_Click(object sender, EventArgs e)
        {
            this.Close();
        }
    }
}

File: VB.NET/LizardKingStudios.Services/LizardKingStudios.Services.SpellcheckEngine/userConfirm.vb
Imports System
Imports System.Windows.Forms
Imports Microsoft.Office.Interop.Word

Partial Friend Class userConfirm
    Inherits Form

    Private _suggestionsList As SpellingSuggestions = Nothing
    Private _selectedValue As String = String.Empty

    Public Sub New()

        InitializeComponent()

    End Sub

    Public WriteOnly Property suggestionsList() As SpellingSuggestions
        Set(ByVal Value As SpellingSuggestions)

            _suggestionsList = Value

            loadSuggestionsList()
        End Set
    End Property

    Public ReadOnly Property selectedValue() As String
        Get
            Return _selectedValue
        End Get
    End Property

    Private Sub loadSuggestionsList()

        If (Not _suggestionsList Is Nothing) Then

            Dim theSuggestion As SpellingSuggestion
            For Each theSuggestion In _suggestionsList
                Suggestions.Items.Add(theSuggestion.Name)
            Next

            Suggestions.SelectedIndex = 0

        End If
    End Sub

    Friend WithEvents Suggestions As System.Windows.Forms.ListBox
    Friend WithEvents selectSuggestion As System.Windows.Forms.Button
    Friend WithEvents cancelForm As System.Windows.Forms.Button

    
    Private Sub selectSuggestion_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles selectSuggestion.Click

        If (Suggestions.SelectedIndex = -1) Then

            MessageBox.Show("You must select a suggestion first!")

        Else

            _selectedValue = Suggestions.SelectedItem.ToString()

            MessageBox.Show("selected: " & _selectedValue)
            Me.Close()
        End If

    End Sub

    Private Sub cancelForm_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cancelForm.Click
        Me.Close()
    End Sub
End Class

